server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -252,24 +252,30 @@ public void write(OutputStream outputStream) throws WebApplicationException
             )
             .header("X-Druid-Query-Id", queryId);
 
-        if (responseContext.get(ResponseContext.CTX_ETAG) != null) {
-          builder.header(HEADER_ETAG, responseContext.get(ResponseContext.CTX_ETAG));
-          responseContext.remove(ResponseContext.CTX_ETAG);
+        if (responseContext.get(ResponseContext.Key.ETAG) != null) {
```

Comment:
Double get looks awkward. It could be
```java
Object entityTag = responseContext.remove(ResponseContext.Key.ETAG);
if (entityTag != null) {
  builder.header(HEADER_ETAG, entityTag);
}
```

Reply:
Nice catch, updated